pandas/io/excel/_base.py

```diff
@@ -453,7 +491,20 @@ def parse(
             else:  # assume an integer if not a string
                 sheet = self.get_sheet_by_index(asheetname)
 
-            data = self.get_sheet_data(sheet, convert_float)
+            get_sheet_data_header = 0 if header is None else header
```

Comment:
Is this required? Looks like there are already checks within the subsequent functions for int values, no?